_posts/2025-09-15-quarkus-a2a-multi-agent-content-creation.adoc

```diff
@@ -0,0 +1,244 @@
+---
+layout: post
+title: 'Multi-Language Agent Collaboration and Interoperability with A2A'
+date: 2025-09-15
+tags: ai a2a
+synopsis: Let's learn how to build a multi-agent system where agents written in different languages and with different LLM frameworks can seamlessly collaborate using A2A.
+author: fjuma
+---
+:imagesdir: /assets/images/posts/quarkus-a2a-multi-agent-content-creation
+
+Building a multi-agent system can involve using different languages to meet specific needs. The https://a2a-protocol.org/latest/[Agent2Agent (A2A) protocol] is an open standard that enables AI agents to communicate and collaborate with one another, regardless of each agent's underlying technology stack.
+
+In this post, we'll see how to create a multi-agent system, where agents written in Java, Python, and TypeScript work together to accomplish a goal: content creation. The multi-agent system uses A2A for communication between the AI agents.
+
+image::ContentCreationDiagram.png[scaledwidth=100%]
+
+== Content Creation Sample
+
+We're going to do a deep dive into the https://github.com/a2aproject/a2a-samples/tree/main/samples/python/hosts/content_creation[Content Creation] sample from the https://github.com/a2aproject/a2a-samples[a2a-samples] project.
+
+This sample showcases a content creation pipeline with a Host agent that acts as the central orchestrator,
+dynamically routing requests to a set of specialized agents.
+
+=== Agents
+
+Here's a quick overview of all the agents in our multi-agent system:
+
+[cols="a,a,a,a",options=header]
+|===
+| Agent | Role | Technology Stack | Description
+
+| *Host*
+| A2A Client
+| Python, Google ADK, A2A Python SDK
+| Serves as the central orchestrator, routing requests to the appropriate agent based on the task at hand.
+
+| *Content Planner*
+| A2A Server
+| Python, Google ADK, A2A Python SDK
+| Receives a high-level content request and creates a detailed content outline.
+
+| *Content Writer*
+| A2A Server
+| Java, Quarkus LangChain4j, A2A Java SDK
+| Generates engaging content from a content outline.
+
+| *Content Editor*
+| A2A Server
+| TypeScript, Genkit, A2A JS SDK
+| Proofreads and polishes the given content.
+|===
+
+Notice that the agents are written in different programming languages and make use of different LLM frameworks.
+
+=== Handling a Content Creation Request
+
+Upon receiving a content creation request from the user, the Host agent breaks down the content creation task
```

Comment:
Can you give names to the agents and reuse these names here?

Reply:
The table and diagrams use the names "Host", "Content Planner", "Content Writer", and "Content Editor". I'll update the references to use `Host`, `Content Planner`, `Content Writer`, and `Content Editor`. Hopefully that's more clear.